src/borkdude/deps.clj

```diff
@@ -225,43 +225,172 @@ For more info, see:
+               "        if (conn instanceof HttpURLConnection)"
+               "           {((HttpURLConnection) conn).setInstanceFollowRedirects(true);}"
+               "        ReadableByteChannel readableByteChannel = Channels.newChannel(conn.getInputStream());"
+               "        FileOutputStream fileOutputStream = new FileOutputStream(args[1]);"
+               "        FileChannel fileChannel = fileOutputStream.getChannel();"
+               "        fileOutputStream.getChannel().transferFrom(readableByteChannel, 0, Long.MAX_VALUE);"
+               "        System.exit(0);"
+               "    } catch (IOException e) {"
+               "        e.printStackTrace();"
+               "        System.exit(1); }}}"))
+    dest-file))
+
+
+(defn- clojure-tools-download-java
+  "Downloads URL file to DEST-ZIP-FILE by invoking `java` with JVM-OPTS
+  on a `.java` program file, and returns true on success. Requires
+  Java 11+ (JEP 330)."
+  [url dest-zip-file jvm-opts]
+  (let [dest-dir (.getCanonicalPath (io/file dest-zip-file ".."))
+        dlr-path (clojure-tools-java-downloader-spit dest-dir)
+        java-cmd (get-java-cmd)
+        success?* (atom true)]
+    (binding [*exit-fn* (fn
+                          ([exit-code] (when-not (= exit-code 0) (reset! success?* false)))
+                          ([exit-code msg] (when-not (= exit-code 0)
+                                             (warn msg)
+                                             (reset! success?* false))))]
+      (shell-command (vec (concat [java-cmd]
+                                  jvm-opts
+                                  [dlr-path url (str dest-zip-file)])))
+      (io/delete-file dlr-path true)
+      @success?*)))
 
 (defn clojure-tools-jar-download
-  "Downloads clojure tools jar into deps-clj-config-dir."
-  [deps-clj-config-dir]
-  (let [dir (io/file deps-clj-config-dir)
-        zip (io/file deps-clj-config-dir "tools.zip")]
-    (.mkdirs dir)
-    (download (format "https://download.clojure.org/install/clojure-tools-%s.zip" @version)
-              zip)
-    (unzip zip (.getPath dir))
-    (.delete zip))
+  "Downloads clojure tools archive in OUT-DIR, if not already there,
+  and extracts in-place the clojure tools jar file and other important
+  files.
+
+  The download is attempted by the following two methods in order:
+
+  1. via java, invoking a java subprocess with
+  JAVA-ARGS-WITH-CLJ-JVM-OPTS options, or
+
+  2. directly from this process.
+
+  It calls `*exit-fn*` if it cannot download the archive, with
+  instructions how to manually download it."
+  [out-dir java-args-with-clj-jvm-opts]
+  (let [{:keys [ct-error-exit-code ct-url-str ct-zip-name]} @clojure-tools-info*
+        dir (io/file out-dir)
+        zip-file (io/file out-dir ct-zip-name)]
+    (when-not (.exists zip-file)
+      (warn "Downloading" ct-url-str "to" (str zip-file))
+      (.mkdirs dir)
+      (or (do (warn "\nAttempting download using " ::java-subprocess-method...
```

Comment:
I think the leading newline can be removed. The auto-resolved keyword  can be removed as well. 

We should only log once that we're attempting the download, regardless of the method, and not cause too much noise imo. 

We can log extra info when `DEPS_CLJ_DEBUG` has been set so when people have problems, we can ask them to set this variable.

Reply:
Done, only msgs before this patch are `warn'ed`, new simpified error messages are only printed out when the new `DEPS_CLJ_DEBUG` env variable is set.